base/promotion.jl

```diff
@@ -299,7 +299,7 @@ function promote_type end
 
 promote_type()  = Bottom
 promote_type(T) = T
-promote_type(T, S, U, V...) = (@inline; promote_type(T, promote_type(S, U, V...)))
+promote_type(T, S, U, V...) = (@inline; promote_type(T, promote_type(S, afoldl(promote_type, U, V...))))
```

Comment:
That is true. I suppose I was being too unnecessarily clever with this one.
```suggestion
promote_type(T, S, U, V...) = (@inline; afoldl(promote_type, T, S, U, V...))
```